src/searchModal.js

```diff
@@ -0,0 +1,207 @@
+        $('.modal-bg').remove();
+    }
+
+    /**
+     * Creates search modal
+     */
+    function initModal() {
+        instance
+        .getElement()
+        .addClass('modal')
+        .on('closed.modal', function() {
+            instance.destroy();
+        })
+        .modal({
+            disableEscape: true,
+            width: $( window ).width(),
+            minHeight: $( window ).height(),
+            modalCloseClass: 'modal-close-left'
+        })
+        .focus();
+    }
+
+    /**
+     * Inits template selectors and sets initial search query on search input
+     */
+    function initUiSelectors() {
+        searchButton = $('.btn-search', instance.getElement());
+        clearButton = $('.btn-clear', instance.getElement());
+        searchInput = $('.search-bar-container input', instance.getElement());
+        searchButton.on('click', search);
+        clearButton.on('click', clear);
+        searchInput.val(config.query);
+    }
+
+    /**
+     * Request search results and manages its results
+     */
+    function search() {
+        const query = searchInput.val();
+        if (query === '') {
+            clear();
+            return;
+        }
+        //throttle and control to prevent sending too many requests
+        const searchHandler = _.throttle(function searchHandlerThrottled(query){
+            if(running === false){
+                running = true;
+                $.ajax({
+                    url : config.url,
+                    type : 'POST',
+                    data :  {query : query},
+                    dataType : 'json'
+                }).done(buildSearchResultsDatatable).complete(function(){
+                    running = false;
+                });
+            }
+        }, 100);
+
+        searchHandler(query);
+    }
+
+    /**
+     * Creates a datatable with search results
+     * @param {object} data - search query results
+     */
+    function buildSearchResultsDatatable(data){
+        //update the section container
+        const $tableContainer = $('<div class="flex-container-full"></div>');
+        const section = $('.content-container', instance.getElement());
+
+        section.empty();
+        section.append($tableContainer);
+        $tableContainer.on('load.datatable', searchResultsLoaded);
+
+        //create datatable
+        $tableContainer.datatable({
+            url: data.url,
+            model : _.values(data.model),
+            emptyText: "No results were found.",
+            labels: {
+                actions: ''
+            },
+            actions : {
+                open : {
+                    label: 'Go to item',
+                    id: 'go-to-item',
+                    action: function openResource(id){
+                        config.events.trigger('refresh', {
+                            uri: id
+                        });
+                        destroyModal();
+                    }
+                }
+            },
```

Comment:
Please use `actions` options as Array

```suggestion
            actions : [
                  {
                    id: 'go-to-item',
                    icon: 'open',
                    label: __('Go to item'),
                    action: function openResource(id){
                        config.events.trigger('refresh', {
                            uri: id
                        });
                        destroyModal();
                    }
                }
            ],
```